ACM4973 update in metrics

cadawson opened the pull request.

cadawson committed:
ACM4973 update in metrics

cadawson committed:
update metrics part 2

cadawson committed:
add metrics 3

zhiweiyin318 commented:
@cadawson  as you asked, do we need to change these name with `mce_` instead of `acm_`. and where can user find these metrics in MCE doc or ACM doc ?  cc @elgnay

cadawson committed:
acm4973 resolve conrefs

cadawson commented:
Hi @zhiweiyin318. I am not sure who makes the naming decision, if it is not your choice. If the metrics are MCE related, then I would think that starting them with ACM would make people think that you have to have ACM to use them. 

The link to staging version of the metrics book is here: https://dxp-docp-prod.apps.ext-waf.spoke.prod.us-west-2.aws.paas.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html-single/health_metrics/index?lb_target=stage.

At the beginning of the health metrics, there is a link to the metrics chronicle. When you select it, you see a list of the compatibility of the metrics: https://github.com/stolostron/metrics-chronicle#application-metrics. I would like to add a direct link on that page, if possible. If not, then there is a link: View a list of metrics for each component in the [metric-chronicle/docs](https://github.com/stolostron/metrics-chronicle/blob/main/docs/).

cadawson committed:
Add server foundation link

cadawson committed:
Add server found metrics

cadawson committed:
Add link to serv found metrics

cadawson commented:
Hi @bjoydeep and @zhiweiyin318 . I'm adding Joydeep to this issue, as it is related to the metrics work that I asked him to  look at as I try to figure out how these metrics are structured. I propose that we close the following PR: https://github.com/stolostron/rhacm-docs/pull/5136 which was to add these metrics to the docs, and instead use this one where we are adding them to the metrics chronicle. Any objections?

zhiweiyin318 commented on docs/acm/component/server-foundation/metrics.md:
```diff
@@ -6,12 +6,12 @@ Nomenclature:
 - `New` implies just added as a part of ACM 2.6 / ACM 2.7
```
all foundation metrics are added in 2.7 except acm_managed_cluster_info, I think we should re-define `New` and the Status of foundation metrics

zhiweiyin318 commented:
@cadawson  discussed with @elgnay , the acm_managed_cluster_info metric  are introduced when there is no MCE. the component to expose these metric is included into MCE when we split MCE from ACM.  and we follow the old name pattern with the prefix `acm_xx`. now it is complicated to rename these metric names since the effect is big.

elgnay commented:
> Hi @bjoydeep and @zhiweiyin318 . I'm adding Joydeep to this issue, as it is related to the metrics work that I asked him to look at as I try to figure out how these metrics are structured. I propose that we close the following PR: [stolostron/rhacm-docs#5136](https://github.com/stolostron/rhacm-docs/pull/5136) which was to add these metrics to the docs, and instead use this one where we are adding them to the metrics chronicle. Any objections?

@cadawson I'm okay to put the foundation metrics into metrics-chronicle repo together with metrics from other components.

cadawson committed:
acm4973 review comments

cadawson committed:
Update new def and status of new

cadawson commented:
Hi @zhiweiyin318, @elgnay, and @bjoydeep. I updated the definition of New and adjusted the status accordingly, indicating that only `acm_managed_cluster_info` is new. 

As for the naming, I thought it might be too late to change the name now, but wanted to raise it. If it was a minor impact item, it might have made good sense to change. It sounds like it is not minor impact. 

Let's see if Joydeep has any feedback. Thank you!

bjoydeep commented:
@cadawson  Minor change needed:
acm_managed_cluster_info, acm_managed_cluster_labels are `NOT` new.
All other metrics are `new`

Exposed and Stable looks good.

zhiweiyin318 commented:
checked with @elgnay  all metrics were exposed in 2.7 except `acm_managed_cluster_info` .  cc @cadawson @bjoydeep

cadawson committed:
update new indicator

cadawson committed:
Updated new status for review comment

cadawson commented:
Thanks @zhiweiyin318 and @bjoydeep. I saw Joydeep's comment and changed it. Then I saw Zhiwei's and changed it back. Please let me know which is correct. It is a pretty easy fix, but we want it to be correct. Thanks!

zhiweiyin318 commented:
@cadawson   I find the story to expose the acm_managed_cluster_labels metrics in 2.7 https://issues.redhat.com/browse/ACM-1795 and the PR  https://github.com/stolostron/clusterlifecycle-state-metrics/pull/99

cadawson commented:
Thanks @zhiweiyin318. Then it sounds like it might be correct as it is now. It is a few hours earlier where @bjoydeep is, so hopefully he will have a chance to review shortly.

bjoydeep commented:
and @cadawson now, since you changed definition of new to 2.8, this looks good: ``New` implies just added as a part of Red Hat Advanced Cluster Management version 2.8 or multicluster engine version 2.3.`

bjoydeep commented:
/lgtm

cadawson commented:
Looks like this was merged automatically. Let me know if you have any last comments @zhiweiyin318. Thanks!

oafischer commented on docs/acm/component/server-foundation/metrics.md:
```diff
@@ -3,15 +3,15 @@
+| acm_managed_cluster_count  | - | Managed cluster count | None |  Not exposed, STABLE |
+| acm_managed_cluster_addon_status_condition | gauge | Managed cluster addon status condition | <table> <tr> <th>Number</th> <th>Label</th> <th>Description</th> </tr> <tr> <td>1</td> <td>`addon_name`</td> <td>The name of the add-on.</td> </tr> <tr> <td>2</td> <td>`managed_cluster_id`</td> <td>The name of the managed cluster to which the add-on belongs.</td> </tr> <tr> <td>3</td> <td>`managed_clus ...
+| acm_managed_cluster_status_condition | gauge | Managed cluster status condition | <table> <tr> <th>Number</th> <th>Label</th> <th>Description</th> </tr> <tr> <td>1</td> <td>`managed_cluster_id`</td> <td>The managed cluster ID is the cluster ID if the managed cluster is an {ocp-short} cluster, otherwise it is the cluster name.</td> </tr> <tr> <td>2</td> <td>`managed_cluster_name`</td> <td>The na ...
+| acm_manifestwork_status_condition | gauge | Manifestwork status condition | <table> <tr> <th>Number</th> <th>Label</th> <th>Description</th> </tr> <tr> <td>1</td> <td>`manifestwork`</td> <td>The name of the manifestwork.</td> </tr> <tr> <td>2</td> <td>`managed_cluster_id`</td> <td>The name of the managed cluster to which the manifestwork belongs.</td> </tr> <tr> <td>3</td> <td>`managed_cluster_ ...
```
Hard to pinpoint the exact spot in this long line but make sure to use "such as" instead of "like" when listing specific examples.

cadawson replied:
Thanks. I picked those up in another PR.
